File: 165-compare-version-numbers/165-compare-version-numbers.py
class Solution(object):
    def compareVersion(self, version1, version2):
        """
        :type version1: str
        :type version2: str
        :rtype: int
        """
        v1, v2 = list(map(int, version1.split('.'))), list(map(int, version2.split('.')))
        
        #making len equal if not
        l1=len(v1)
        l2=len(v2)
        
        if(l1>l2):
            for _ in range(l1-l2):
                v2.append(0)
                
        elif(l1<l2):
            for _ in range(l2-l1):
                v1.append(0)
                
        for i in range (max(l1,l2)):
            if(v1[i]>v2[i]):
                return 1
            
            if(v1[i]<v2[i]):
                return -1
        
        return 0

File: 165-compare-version-numbers/165-compare-version-numbers.cpp
class Solution {
public:
    
    void solve(string &version, vector<int> &v){
        
        version+='.';
        
        string temp="";
        for(auto c:version){
            
            if(c=='.'){
                v.push_back(stoi(temp));
                temp="";
            }else{
                temp+=c;
            }
            
        }
        
        
    }
    
    int compareVersion(string version1, string version2) {
        
        vector<int> v1,v2;
        
        solve(version1,v1);
        solve(version2,v2);
        
        while(v1.size()<v2.size()) v1.push_back(0);
        while(v2.size()<v1.size()) v2.push_back(0);
        
        for(int i=0;i<v1.size();i++){
            
            if(v1[i]>v2[i]) return 1;
            
            if(v1[i]<v2[i]) return -1;
            
        }    
        return 0;
    }
};